src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducer().get();
+        }
+        if (type == Truck.class) {
+            machines = new TruckProducer().get();
+        }
+        if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
```

Comment:
services class rely on should be private final variables on class level not to be recreated each time we call some method